### PR TITLE
Clean up reading/writing of labeling configuration in vector layers

### PR DESCRIPTION
**Pull request by wonder-sk**
Historically the configuration used to be stored in layer's custom properties, but that does not scale
beyond simple rendering and so rule-based labeling introduced storage of configuration natively in XML elements. That left us with two different ways of reading/writing labeling configurations. This work makes all configuration to use native XML elements.

To keep compatibility of 2.x projects, reading of configuration from custom properties is preserved.

This commit also adds Python APIs for direct manipulation of labeling configuration through vector layer's setLabeling() and labeling() calls.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Oh hell yes! Those duplicate labeling methods have frustrated me SO much in the past.

**Comment by wonder-sk** [comment]
This should be now ready to go...

File: src/core/qgspallabeling.cpp [hunk 1]
```diff
@@ -678,101 +668,8 @@ void QgsPalLayerSettings::readFromLayer( QgsVectorLayer *layer )
...
-  layer->setCustomProperty( QStringLiteral( "labeling/enabled" ), enabled );
-  layer->setCustomProperty( QStringLiteral( "labeling/drawLabels" ), drawLabels );
-
-  mFormat.writeToLayer( layer );
-
-  // text style
-  layer->setCustomProperty( QStringLiteral( "labeling/fieldName" ), fieldName );
-  layer->setCustomProperty( QStringLiteral( "labeling/isExpression" ), isExpression );
-  layer->setCustomProperty( QStringLiteral( "labeling/previewBkgrdColor" ), previewBkgrdColor.name() );
-  QDomDocument doc( QStringLiteral( "substitutions" ) );
```
Comment (nyalldawson) [inline src/core/qgspallabeling.cpp]: I'm quite relieved that no-one can see this bit of code anymore ;)

**Review by nyalldawson — APPROVED** [review]

File: src/core/qgspallabeling.cpp (hunk 1 above)
Comment (wonder-sk) [inline src/core/qgspallabeling.cpp]: haha - we will keep those bits in our memories!

**Pull request merged**